File: Bailaconmigo/src/main/java/org/example/bailaconmigo/DTOs/GeographicReportDto.java
package org.example.bailaconmigo.DTOs;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class GeographicReportDto {
    private Long totalUsers;
    private Long totalDancers;
    private Integer totalCities;
    private Integer totalCountries;
    private List<EnhancedGeographicDto> topCities;
    private List<EnhancedGeographicDto> topCountries;
    private List<CityStyleRankingDto> cityStyleRankings;
}

File: frontend/src/app/models/geographic-report-dto.ts
import { CityStyleRankingDto } from "./city-style-ranking-dto";
import { EnhancedGeographicDto } from "./enhanced-geographic-dto";

export interface GeographicReportDto {
  totalUsers: number;
  totalDancers: number;
  totalCities: number;
  totalCountries: number;
  topCities: EnhancedGeographicDto[];
  topCountries: EnhancedGeographicDto[];
  cityStyleRankings: CityStyleRankingDto[];
}